### PR TITLE
Allow to build without any warning with NDEBUG defined + Bugfix

### PR DESCRIPTION
**Pull request by benjarobin**

### CODE REVIEW

**Comment by benjarobin** [comment]
Since I found 2 bugs in the code by adding with environment  `CFLAGS` variable `-DNDEBUG=1`, should we remove the following bloc from `meson.build` (and since now there is no longer any warning with `NDEBUG`) ?

```
if get_option('b_ndebug') == 'true'
        # With asserts disabled with get a bunch of warnings about variables which
        # are used only in the asserts. This is not useful at all, so let's just silence
        # those warnings.
        basic_disabled_warnings += [
                '-Wno-unused-variable',
                '-Wno-unused-but-set-variable',
        ]
endif
```

File: src/shared/resize-fs.c [hunk 1]
```diff
@@ -42,6 +41,7 @@ int resize_fs(int fd, uint64_t sz, uint64_t *ret_size) {
                         *ret_size = u * sfs.f_bsize;
 
         } else if (is_fs_type(&sfs, BTRFS_SUPER_MAGIC)) {
+                _unused_ int r;
```
Comment (keszybz) [inline src/shared/resize-fs.c]: Please use xsprintf instead. Then `r` will not be needed at all.

File: src/import/import-fs.c [hunk 2]
```diff
@@ -172,8 +172,8 @@ static int import_fs(int argc, char *argv[], void *userdata) {
         progress.limit = (RateLimit) { 200*USEC_PER_MSEC, 1 };
 
         /* Hook into SIGINT/SIGTERM, so that we can cancel things then */
-        assert(sigaction(SIGINT, &sa, &old_sigint_sa) >= 0);
-        assert(sigaction(SIGTERM, &sa, &old_sigterm_sa) >= 0);
+        assert_se(sigaction(SIGINT, &sa, &old_sigint_sa) >= 0);
+        assert_se(sigaction(SIGTERM, &sa, &old_sigterm_sa) >= 0);
```
Comment (keszybz) [inline src/import/import-fs.c:176]: Yikes!

**Comment by keszybz** [comment]
Maybe include another commit to drop that block in meson.build ?

**Comment by benjarobin** [comment]
> Maybe include another commit to drop that block in meson.build ?

Done, thanks for the review

**Comment by keszybz** [comment]
CentOS CI (Arch in KVM) is stuck after 11h, let's not wait.

**Pull request merged**